staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -637,6 +637,10 @@ func shouldUpdateFinalizerOrphanDependents(e *Store, accessor metav1.Object, opt
 	shouldOrphan = false
 	// Get default orphan policy from this REST object type
 	if gcStrategy, ok := e.DeleteStrategy.(rest.GarbageCollectionDeleteStrategy); ok {
+		if gcStrategy.DefaultGarbageCollectionPolicy() == rest.Unsupported {
+			// return true to force clear the GC policy if needed, and false to indicate that we should NOT orphan
+			return true, false
```

Comment:
nit: shouldUpdate not necessary.

Can't we drop this `shouldUpdate` value from both funcs and add some more clever logic to `shouldUpdateFinalizers` to detect changes in the finalizers? With you change this gets even more complicated.